extensions/data-transfer/portability-data-transfer-google/src/main/java/org/dataportabilityproject/datatransfer/google/photos/GooglePhotosExporter.java

```diff
@@ -152,45 +143,43 @@ public GooglePhotosExporter(GoogleCredentialFactory credentialFactory) {
 
   private ExportResult<PhotosContainerResource> exportPhotos(
       TokensAndUrlAuthData authData, String albumId, Optional<PaginationData> paginationData) {
-
-    int startItem = 1;
+    Optional<String> paginationToken = Optional.empty();
     if (paginationData.isPresent()) {
       String token = ((StringPaginationToken) paginationData.get()).getToken();
       Preconditions.checkArgument(
           token.startsWith(PHOTO_TOKEN_PREFIX), "Invalid pagination token " + token);
-      startItem = Integer.parseInt(token.substring(PHOTO_TOKEN_PREFIX.length()));
+      paginationToken = Optional.of(token.substring(PHOTO_TOKEN_PREFIX.length()));
     }
 
-    URL photosUrl;
-    AlbumFeed photoFeed;
+    MediaItemSearchResponse mediaItemSearchResponse;
 
     try {
-      photosUrl = new URL(String.format(URL_PHOTO_FEED_FORMAT, albumId, startItem, maxResults));
-      photoFeed = getOrCreatePhotosService(authData).getFeed(photosUrl, AlbumFeed.class);
-
-    } catch (ServiceException | IOException e) {
+      mediaItemSearchResponse = getOrCreatePhotosInterface(authData).listAlbumContents(albumId, paginationToken);
+    } catch (IOException e) {
       return new ExportResult<>(ResultType.ERROR, e.getMessage());
     }
 
     PaginationData nextPageData = null;
-    List<GphotoEntry> entries = photoFeed.getEntries();
-    if (entries.size() == maxResults) {
-      int nextPageStart = startItem + maxResults;
-      nextPageData = new StringPaginationToken(PHOTO_TOKEN_PREFIX + nextPageStart);
+    GoogleMediaItem[] googleMediaItems = mediaItemSearchResponse.getMediaItems();
```

Comment:
Also here why create a bunch of other variables instead of just iterating over mediaItemSearchResponse.getMediaItems()?